File: src/assets/code.cs
// CREATING
DateTime dt;
dt = DateTime.UtcNow;
dt = DateTime.Now;
dt = new DateTime(2023, 1, 2, 8, 0, 0, DateTimeKind.Utc);
dt = new DateTime(2023, 1, 2, 0, 0, 0, DateTimeKind.Local);
dt = DateTimeOffset.FromUnixTimeSeconds(1672646400).DateTime.ToUniversalTime();
dt = DateTimeOffset.FromUnixTimeSeconds(1672646400).DateTime.ToLocalTime();
dt = DateTime.Parse("2023-01-02T08:00:00.000Z").ToUniversalTime();
dt = DateTime.Parse("2023-01-02T08:00:00.000Z");

// TIMEZONE
Console.WriteLine(dt.Kind); // 'Local' or 'Utc'

// EXPORTING
Console.WriteLine(dt);                                          // =>  1/2/2023 12:00:00 AM
Console.WriteLine(new DateTimeOffset(dt).ToUnixTimeSeconds());  // =>  1672646400
Console.WriteLine(dt.ToUniversalTime().ToString("o"));          // =>  2023-01-02T08:00:00.0000000Z
// Formatting: https://learn.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings
Console.WriteLine(dt.ToString("d-MMM-yyyy h:mm tt"));           // =>  2-Jan-2023 12:00 AM
Console.WriteLine(dt.ToString("dddd MMMM d, yyyy h:mm:ss tt")); // =>  Sunday January 2, 2023 12:00:00 AM
Console.WriteLine(dt.ToString("yyyy-MM-dd_HH:mm:ss"));          // =>  2023-01-02_00:00:00

// COMPONENTS
int year = dt.Year;
int month = dt.Month;
int day = dt.Day;
int hour = dt.Hour;
int minute = dt.Minute;
int second = dt.Second;
DayOfWeek day_of_week = dt.DayOfWeek; // Enum

// MODIFYING
DateTime dt2 = dt; // Cloning
dt = dt.AddDays(2); // Add 2 days
dt = dt.AddHours(3); // Add 3 hours

File: src/assets/code.py
from datetime import datetime, timezone, timedelta

# CREATING
local_tz = datetime.now(timezone.utc).astimezone().tzinfo
dt = datetime.now(timezone.utc)
dt = datetime.now(local_tz)
dt = datetime(2023, 1, 2, 8, 0, 0, 0, timezone.utc)
dt = datetime(2023, 1, 2, 0, 0, 0, 0, local_tz)
dt = datetime.fromtimestamp(1672646400, timezone.utc)
dt = datetime.fromtimestamp(1672646400, local_tz)
dt = datetime.strptime("2023-01-02T08:00:00.000Z", "%Y-%m-%dT%H:%M:%S.%f%z")
dt = datetime.strptime("2023-01-02T08:00:00.000Z", "%Y-%m-%dT%H:%M:%S.%f%z").astimezone(local_tz)

# TIMEZONE STUFF
print(dt.tzinfo) # 'Pacific Standard Time' or 'UTC'

# EXPORTING
print(dt)                                                            # => 2023-01-02 00:00:00-08:00
print(int(datetime.timestamp(dt)))                                   # => 1672646400
print(dt.astimezone(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.%fZ")) # => 2023-01-02T08:00:00.000000Z
# Formatting: https://strftime.org/ (note: zero-padding not available on windows)
print(dt.strftime("%d-%b-%Y %I:%M %p"))                              # => 02-Jan-2023 12:00 AM
print(dt.strftime("%A %B %d, %Y %I:%M:%S %p"))                       # => Monday January 02, 2023 12:00:00 AM
print(dt.strftime("%Y-%m-%d_%H:%M:%S"))                              # => 2023-01-02_00:00:00

# COMPONENTS
year = dt.year
month = dt.month
day = dt.day
hour = dt.hour
minute = dt.minute
second = dt.second
day_of_week = dt.weekday() # [0-6], 0 == Monday

# MODIFYING
dt2 = dt # Cloning
dt = dt + timedelta(days=2) # Add 2 days
dt = dt + timedelta(hours=3) # Add 3 hours